core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -293,13 +292,16 @@ private[spark] class TaskSchedulerImpl(
     // Also track if new executor is added
     var newExecAvail = false
     for (o <- offers) {
-      executorIdToHost(o.executorId) = o.host
-      executorIdToTaskCount.getOrElseUpdate(o.executorId, 0)
       if (!executorsByHost.contains(o.host)) {
         executorsByHost(o.host) = new HashSet[String]()
+      }
+      if (!executorIdToHost.contains(o.executorId)) {
```

Comment:
It looks like this and #13603 are tangled a bit, since there's a test here that was introduced in that change, but these lines were also changed in #13603.  I think the changes on this line belong in this PR -- if that's correct, can you revert the changes in this PR that are also in #13603, so we can merge this one first?